File: CS/DataObjects/SimpleExampleContext.cs
using System.Linq;
using InterLinq;

namespace DataObjects
{
    /// <summary>
    /// This is a context for easier access to the <see cref="IQueryHandler"/>.
    /// </summary>
    public class SimpleExampleContext : InterLinqContext
    {
        public SimpleExampleContext(IQueryHandler queryHandler)
            : base(queryHandler) {
        }

        public IQueryable<WpfServerSideGridTest> Objects {
            get { return QueryHander.Get<WpfServerSideGridTest>(); }
        }

    }
}

File: VB/DataObjects/SimpleExampleContext.vb
Imports Microsoft.VisualBasic
Imports System.Linq
Imports InterLinq

Namespace DataObjects
	''' <summary>
	''' This is a context for easier access to the <see cref="IQueryHandler"/>.
	''' </summary>
	Public Class SimpleExampleContext
		Inherits InterLinqContext
		Public Sub New(ByVal queryHandler As IQueryHandler)
			MyBase.New(queryHandler)
		End Sub

		Public ReadOnly Property Objects() As IQueryable(Of WpfServerSideGridTest)
			Get
				Return QueryHander.Get(Of WpfServerSideGridTest)()
			End Get
		End Property

	End Class
End Namespace
